### PR TITLE
fix(locale): filter inappropriate words in the en locale

### PR DESCRIPTION
**Pull request by rcfox**
Addressing #1631

### CODE REVIEW

**Comment by ST-DDT** [comment]
IMO the English locale is big enough that we can be quite thoroughly with the removals.
Some of the words don't feel offending by themselves, but I agree, that they might cause irritation even by unawareness of their (original) meaning. (Especially when mixed with random words e.g. any adjective+color.)

**Comment by ST-DDT** [comment]
**Team decision**

We will check:

- each of your suggested removals once your PR is complete, for some we aren't sure yet, but the general direction is good
- the en locale for additional entries that needs to be removed
- whether there are list's of words in the internet that can be considered offensive/unprofessional and use it for additional filtering

**Comment by rcfox** [comment]
This PR is ready for review now. I do realize I probably went a little overboard on some of the words, so it's unlikely I'll have any arguments about words to reinstate. (With such long lists, I sort of had to turn part of my brain off and go by a gut feeling.)

I did initially try a using a list of bad words meant for moderation software, but felt they lacked nuance.

**Review by ST-DDT — DISMISSED** [review]

**Comment by ST-DDT** [comment]
Please run `pnpm run test -u` to update the test snapshots.

**Comment by rcfox** [comment]
Rebased on latest 'next' branch. Updated test snapshots. Squashed the commits into a single commit.

**Review by ST-DDT — DISMISSED** [review]

**Review by Shinigami92 — COMMENTED** [review]
Looks like there is a merge conflict with `test/__snapshots__/system.spec.ts.snap`
Please rebase your PR and solve it

---

I would like to _softly_ "enforce" a required approval from @xDivisionByZerox for this PR

File: src/locales/en/word/adjective.ts [hunk 1]
```diff
@@ -634,7 +630,6 @@ export default [
   'known',
   'kooky',
   'kosher',
-  'lame',
```
Comment (Shinigami92) [inline src/locales/en/word/adjective.ts:637]: :thinking:

File: src/locales/en/word/adjective.ts [hunk 2]
```diff
@@ -776,7 +771,6 @@ export default [
   'nutritious',
   'nutty',
   'obedient',
-  'obese',
```
Comment (Shinigami92) [inline src/locales/en/word/adjective.ts:779]: :thinking:

File: src/locales/en/word/adjective.ts [hunk 3]
```diff
@@ -1172,7 +1165,6 @@ export default [
   'trustworthy',
   'trusty',
   'truthful',
-  'tubby',
```
Comment (Shinigami92) [inline src/locales/en/word/adjective.ts:1175]: :thinking:

File: src/locales/en/word/adverb.ts [hunk 4]
```diff
@@ -200,7 +200,6 @@ export default [
   'quaintly',
   'quarrelsomely',
   'queasily',
-  'queerly',
```
Comment (Shinigami92) [inline src/locales/en/word/adverb.ts:203]: :thinking:

File: src/locales/en/word/noun.ts [hunk 5]
```diff
@@ -11,7 +11,6 @@ export default [
   'ability',
   'abnormality',
   'abolishment',
-  'abortion',
```
Comment (Shinigami92) [inline src/locales/en/word/noun.ts:14]: :thinking:

File: src/locales/en/word/noun.ts [hunk 6]
```diff
@@ -514,7 +509,6 @@ export default [
   'bee',
   'beech',
   'beef',
-  'beer',
```
Comment (Shinigami92) [inline src/locales/en/word/noun.ts:517]: :thinking:

File: src/locales/en/word/noun.ts [hunk 7]
```diff
@@ -581,7 +574,6 @@ export default [
   'bite',
   'bitten',
   'bitter',
-  'black',
```
Comment (Shinigami92) [inline src/locales/en/word/noun.ts:584]: :thinking:

File: src/locales/en/word/noun.ts [hunk 8]
```diff
@@ -614,7 +606,6 @@ export default [
   'blow',
   'blowgun',
   'blowhole',
-  'blue',
```
Comment (Shinigami92) [inline src/locales/en/word/noun.ts:617]: :thinking:

File: src/locales/en/word/noun.ts [hunk 9]
```diff
@@ -700,9 +689,7 @@ export default [
   'bran',
   'branch',
   'brand',
-  'brandy',
```
Comment (Shinigami92) [inline src/locales/en/word/noun.ts:703]: :thinking:

File: src/locales/en/word/noun.ts [hunk 10]
```diff
@@ -746,7 +731,6 @@ export default [
   'brother',
   'brother-in-law',
   'brow',
-  'brown',
```
Comment (Shinigami92) [inline src/locales/en/word/noun.ts:749]: :thinking:

File: src/locales/en/word/noun.ts [hunk 11]
```diff
@@ -1970,7 +1938,6 @@ export default [
   'dust',
   'duster',
   'duty',
-  'dwarf',
```
Comment (Shinigami92) [inline src/locales/en/word/noun.ts:1973]: :thinking:

File: src/locales/en/word/noun.ts [hunk 12]
```diff
@@ -2896,7 +2854,6 @@ export default [
   'heron',
   'herring',
   'hesitation',
-  'heterosexual',
```
Comment (Shinigami92) [inline src/locales/en/word/noun.ts:2899]: I'm splitted 🙃

File: src/locales/en/word/noun.ts [hunk 13]
```diff
@@ -2934,11 +2891,8 @@ export default [
   'homeownership',
   'hometown',
   'homework',
-  'homicide',
   'homogenate',
   'homonym',
-  'homosexual',
```
Comment (Shinigami92) [inline src/locales/en/word/noun.ts:2940]: I'm splitted 🙃

File: src/locales/en/word/noun.ts [hunk 14]
```diff
@@ -4501,7 +4437,6 @@ export default [
   'pineapple',
   'pinecone',
   'ping',
-  'pink',
```
Comment (Shinigami92) [inline src/locales/en/word/noun.ts:4504]: :thinking:

File: src/locales/en/word/noun.ts [hunk 15]
```diff
@@ -4820,7 +4754,6 @@ export default [
   'purchase',
   'puritan',
   'purity',
-  'purple',
```
Comment (Shinigami92) [inline src/locales/en/word/noun.ts:4823]: :thinking:

File: src/locales/en/word/noun.ts [hunk 16]
```diff
@@ -4961,7 +4891,6 @@ export default [
   'recreation',
   'recruit',
   'rectangle',
-  'red',
```
Comment (Shinigami92) [inline src/locales/en/word/noun.ts:4964]: :thinking:

**Comment by rcfox** [comment]
These merge conflicts are going to keep cropping up as other changes are merged in. I'm going to wait until this ticket is otherwise ready to be merged and handle conflicts then.

**Review by Shinigami92 — COMMENTED** [review]
Okay, I added some ":thinking:" to the first two files

I think we need to add a sentence / explanation to a file why some words are not allowed

Maybe the contrib guidelines, maybe a comment at the top of the file?

I'm not sure about the colors, we could say: no colors => use colors module for that and I'm fine
I'm not sure about some words like "lame" as yeah they might be like "boring" but I don't think they need to be totally omitted

"obese", "heterosexual", "homosexual" => I think it is more inclusive to have these words in our set as the otherway around

There are some alcoholic words :thinking: maybe we need to transfer these to the requested food module (like colors)

"dwarf" => it is just a fantasy word :thinking: 

I understand why to remove sexual affected words, but not sure where to draw the line between normal talk and "above-18" words

**Comment by ST-DDT** [comment]
> I think we need to add a sentence / explanation to a file why some words are not allowed

> "obese", "heterosexual", "homosexual" => I think it is more inclusive to have these words in our set as the otherway around

> I understand why to remove sexual affected words, but not sure where to draw the line between normal talk and "above-18" words

https://github.com/faker-js/faker/blob/db6b5551ecc0bc687f755a7d9fb5a52dad134e95/CODE_OF_CONDUCT.md?plain=1#L5

Since there is no need to include any words regarding

> age, body size, disability, ethnicity, gender identity and expression, level of experience, education, socio-economic status, nationality, personal appearance, race, religion, or sexual identity and orientation

we should avoid adding any of them, because we still have plenty of other words. 
(Just to prevent accidentally offending someone by generating `good ___` or `bad ___`.)
The list of words is supposed to free of any intent to avoid creating offense by joining valuing words to words with intent.
It is easier to remove words referring to specific persons, ... than removing words that contain value.

> Maybe the contrib guidelines

Maybe, I just don't know how to exactly formulate that.

> maybe a comment at the top of the file?

This is somewhat special to the word category in general and not a single file.

**Review by terryknowlton — DISMISSED** [review]
:+1:  We appreciate the cleanup.

**Review by ejcheng — DISMISSED** [review]

**Review by Shinigami92 — DISMISSED** [review]
We also discussed to remove entries from locales where a file contains more then 1k entries (like names or nouns) so files are not getting bigger and bigger
1k entries are more then enough for normal use cases

**Review by ST-DDT — APPROVED** [review]

**Review by xDivisionByZerox — APPROVED** [review]
I'll approve this for two reasons, even tho I'm not happy with some words being removed:
1. It's reasonable to have this in the alpha release, as people are actually interested in this
2. As @ST-DDT said: We have so many words in the `en` locale, it doesn't matter if we get rid of some

**Review by Shinigami92 — APPROVED** [review]

**Pull request merged**